x-pack/solutions/observability/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -300,7 +304,7 @@ export const createCustomThresholdExecutor = ({
     alertsClient.setAlertLimitReached(hasReachedLimit);
     const recoveredAlerts = alertsClient.getRecoveredAlerts() ?? [];
 
-    let groupingObjectForRecovered: Record<string, unknown> = {};
+    let groupingObjectForRecovered: Record<string, any> = {};
```

Comment:
what do you think about using `kibana.alert.grouping` for `context.grouping` variable in this PR as I suggested in https://github.com/elastic/kibana/issues/219995#issuecomment-2858688344? (assuming we use rule evaluation results to get group values)